x-pack/plugins/observability_solution/apm/server/routes/assistant_functions/get_observability_alert_details_context/index.ts

```diff
@@ -76,70 +78,93 @@ export async function getObservabilityAlertDetailsContext({
     }),
   ]);
 
+  async function handleError<T>(cb: () => Promise<T>): Promise<T | undefined> {
+    try {
+      return await cb();
+    } catch (error) {
+      logger.error('Error while fetching observability alert details context');
+      logger.error(error);
+      return;
```

Comment:
Should you throw the error to let the client know?

Reply:
Isn't it the same as wrapping the requests bellow in a single `try{}catch`?

Reply:
> Isn't it the same as wrapping the requests bellow in a single `try{}catch`?

Yes, `handleError` is just a shorthand for catching errors.